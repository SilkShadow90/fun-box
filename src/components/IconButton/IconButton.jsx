import React from 'react';
import Icon from '@material-ui/core/Icon/Icon';
import Button from '@material-ui/core/Button/Button';
import PropTypes from 'prop-types';

const IconButton = React.memo((props) => {
  const {
    className,
    color,
    icon,
    size,
    onClick,
  } = props;

  return (
    <Button
      variant="contained"
      color={color}
      className={className}
      onClick={onClick}
    >
      <Icon fontSize={size}>{icon}</Icon>
    </Button>
  );
});

IconButton.propTypes = {
  className: PropTypes.string,
  icon: PropTypes.string,
  onClick: PropTypes.func,
  color: PropTypes.string,
  size: PropTypes.string,
};

IconButton.defaultProps = {
  className: undefined,
  icon: 'error',
  onClick: undefined,
  size: 'default',
  color: 'primary',
};

export default IconButton;
