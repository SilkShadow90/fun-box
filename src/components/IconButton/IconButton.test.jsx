import React from 'react';
import Renderer from 'react-test-renderer';

import IconButton from './IconButton';

describe('check IconButton component', () => {
  const renderer = Renderer.create(
    <IconButton
      color="secondary"
      icon="keyboard_arrow_left"
      size="small"
    />,
  );

  const button = renderer.toJSON();

  test('check expect match type', () => {
    expect(button.type).toBe('button');
  });

  test('check expect match children', () => {
    expect(button.children[0].type).toBe('span');
    expect(button.props.className).toMatch('Secondary');
    expect(button.children[0].children[0].props.className).toMatch('fontSizeSmall');
    expect(button.children[0].children[0].children[0]).toBe('keyboard_arrow_left');
    expect(button.children[1].type).toBe('span');
  });

  test('check expect match snapshot', () => {
    expect(button).toMatchSnapshot();
  });
});
