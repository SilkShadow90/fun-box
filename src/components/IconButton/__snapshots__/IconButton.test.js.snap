// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check IconButton component check expect match snapshot 1`] = `
<button
  className="MuiButtonBase-root-27 MuiButton-root-1 MuiButton-contained-12 MuiButton-containedSecondary-14 MuiButton-raised-15 MuiButton-raisedSecondary-17"
  disabled={false}
  onBlur={[Function]}
  onClick={[Function]}
  onContextMenu={[Function]}
  onFocus={[Function]}
  onKeyDown={[Function]}
  onKeyUp={[Function]}
  onMouseDown={[Function]}
  onMouseLeave={[Function]}
  onMouseUp={[Function]}
  onTouchEnd={[Function]}
  onTouchMove={[Function]}
  onTouchStart={[Function]}
  tabIndex="0"
  type="button"
>
  <span
    className="MuiButton-label-2"
  >
    <span
      aria-hidden="true"
      className="material-icons MuiIcon-root-30 MuiIcon-fontSizeSmall-37"
    >
      keyboard_arrow_left
    </span>
  </span>
  <span
    className="MuiTouchRipple-root-39"
  />
</button>
`;
